File: Java/leetcode75/src/test/java/org/phoenix/leetcode/challenges/Problem29_DeleteTheMiddleNodeOfALinkedListTest.java
package org.phoenix.leetcode.challenges;

import org.junit.jupiter.api.Test;
import org.phoenix.leetcode.challenges.Problem29_DeleteTheMiddleNodeOfALinkedList.ListNode;

import static org.junit.jupiter.api.Assertions.assertEquals;

class Problem29_DeleteTheMiddleNodeOfALinkedListTest {

    private final Problem29_DeleteTheMiddleNodeOfALinkedList test = new Problem29_DeleteTheMiddleNodeOfALinkedList();

    @Test
    void deleteMiddle() {
        ListNode head = new ListNode(1);
        head.next = new ListNode(3);
        head.next.next = new ListNode(4);
        head.next.next.next = new ListNode(7);
        head.next.next.next.next = new ListNode(1);
        head.next.next.next.next.next = new ListNode(2);
        head.next.next.next.next.next.next = new ListNode(6);

        assertEquals(1, test.deleteMiddle(head).next.next.next.val);
    }
}

File: JavaScript/leetcode75/src/test/Problem29_DeleteTheMiddleNodeOfALinkedList.test.js
const deleteMiddle = require("../main/Problem29_DeleteTheMiddleNodeOfALinkedList");

function ListNode(val, next) {
  this.val = val === undefined ? 0 : val;
  this.next = next === undefined ? null : next;
}

test("should return list without middle", () => {
  let list = new ListNode(1, 3);
  list.next = new ListNode(3, 4);
  list.next.next = new ListNode(4, 7);
  list.next.next.next = new ListNode(7, 1);
  list.next.next.next.next = new ListNode(1, 2);
  list.next.next.next.next.next = new ListNode(2, 6);
  expect(deleteMiddle(list).next.next.next.val).toBe(1);
});
